db/infinispan/src/main/java/org/commonjava/indy/infinispan/data/InfinispanStoreDataManager.java

```diff
@@ -144,4 +159,78 @@ protected ArtifactStore putArtifactStoreInternal( StoreKey storeKey, ArtifactSto
         return stores.put( storeKey, store );
     }
 
+    @Override
+    public Set<Group> affectedBy( final Collection<StoreKey> keys )
+    {
+        final Set<Group> groups = new HashSet<>();
+        for ( StoreKey key : keys )
+        {
+            Set<StoreKey> affected = affectedByStores.get( key );
+            if ( affected != null )
+            {
+                affected = affected.stream().filter( k -> k.getType() == group ).collect( Collectors.toSet() );
```

Comment:
Okay, I think this won't work as expected. We have the potential for arbitrarily nested groups, which means we might need to lookup from the cache an arbitrary number of times (remember, we decided not to store the fully denormalized affected-by relationship with intermediary groups removed, but instead tolerate arbitrary levels of nesting with direct-lookup from the cache for each nesting level).

To handle this, I think we need a toProcess / processed setup for these groups. The difference is that we don't have to do a full table scan when the process starts, instead doing affected-by cache lookups as needed for specific keys.

**EDIT:** I see below you're implementing a fully denormalized affected-by relationship, which simplifies this retrieval process. However, I wonder if it opens us up to a lot more complexity in the update process?